### ISSUE TITLE
Feature Request: Change result buffer, save, propagate changes to db

### ISSUE THREAD

**Issue by thyseus** [kristijanhusak/vim-dadbod-ui#233]
I did not find this (kick-ass!) Feature Request anywhere else neither i do have enough vim programming skills so i just want to write down that idea here:

When i do a query `select id, name from users` and get a results like 

```
|id | name |
| 1 |  Herbert |
```

i want to EDIT the result buffer, save it, and let DADBOT propagate the changes via `update users set name =  <> where id = <>`.

This would require a primary key and i think this *hard* to implement.

anyone up for a challenge ? :) Thanks a lot !

**Comment by thyseus** [comment]
Of course, removing a line will create a `delete` query; insert a line a `insert into` query... how awesome would that be ?

**Comment by wstewarttennes** [comment]
+1 here although I'm sure this is a complicated build out

**Comment by tinytownsoftware** [comment]
I've been looking for an open source project to work on and I do miss JetBrains DataGrip and the way it did this very thing. So I may just sink my teeth into this.

**Comment by thyseus** [comment]
Would be absolute awesome; out of curiosity: 

vim-dadbod-ui seems to be written in VimScript which is kind of deprecated nowadays;

a lua implementation would be the way to go;

then: i think also some of the necessary work would need to be done in the vim-dadbod repository also.

Whats your opinion ?

**Comment by thyseus** [comment]
I could not find anything more "modern" than dadbod for (neo)vim in the net.

 I think a complete rewrite from scratch would be necessary; there are some really nice CLI UI things like rainfrog (rust) and some written in python, but none of them are integrated into vim.
